### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by MisterDA**

### CODE REVIEW

**Comment by hannesm** [comment]
Thanks for your changes, from what I can tell the issue is changes in 4.4.0. If you apply the following patch, it should work:

```OCaml
--- a/mirage/config.ml
+++ b/mirage/config.ml
@@ -202,10 +202,18 @@ let enable_metrics =
   let doc = Key.Arg.info ~doc:"Enable metrics reporting" [ "metrics" ] in
   Key.(create "metrics" Arg.(flag doc))
 
+type i0 = I0
+let i0 = Functoria.Type.v I0
+let no0 = Functoria.impl "Int" job
+
+type n1 = N1
+let n1 = Functoria.Type.v N1
+let noop1 = Functoria.impl "Set.Make" (job @-> job)
+
 let optional_monitoring time pclock stack =
   if_impl (Key.value enable_metrics)
     (mirage_monitoring $ time $ pclock $ stack)
-    noop
+    (noop1 $ no0)
 
 let () =
   register "www"
```

**Comment by MisterDA** [comment]
Yes, I was going to ask you :) wouldn't have figured that out!

File: Dockerfile [hunk 1]
```diff
@@ -2,9 +2,9 @@ FROM ocaml/opam:debian-12-ocaml-4.14
 RUN sudo apt-get update && sudo apt-get install autoconf automake -y --no-install-recommends
 RUN mkdir -p /home/opam/www/mirage
 WORKDIR /home/opam/www
-RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam && cd ~/opam-repository && git pull origin master && git reset --hard 297ab9ebb38254eee222f5bb4365ae80baa9caa1 && opam update
-RUN opam pin git+https://github.com/tmattio/opam-tailwindcss#a4c0ee9b9a80d44c4a953b07dc29e4aa065adf57
-RUN opam repo add mirage-dev git+https://github.com/mirage/mirage-dev.git#842c55556ffd0950d21141d6ab99e52a8d88a50f
+RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam && cd ~/opam-repository && git pull origin master && git reset --hard 30b1b97d735732e40996cf2e6b06d478ac40633f && opam update
+RUN opam pin tailwindcss.dev https://github.com/tmattio/opam-tailwindcss/archive/3e60fc32bbcf82525999d83ad0f395e16107026b.tar.gz
+RUN opam repo add mirage-dev git+https://github.com/mirage/mirage-dev.git#749c02302f8f15e609332edbe827541558554a80
```
Comment (hannesm) [inline Dockerfile:7]: this we should not need (famous last words) at all ;)

**Comment by hannesm** [comment]
> Yes, I was going to ask you :) wouldn't have figured that out!

Maybe the mirage tool should be relaxed to allow some jobs without arguments (maybe exactly the noop job) -- but the mirage repository is currently in unclear state -- the main branch isn't too usable, we could continue on the 4.4 branch...

**Comment by MisterDA** [comment]
Unfortunately the patch for Dream that was pinned seem not to have been upstreamed.

**Comment by MisterDA** [comment]
Erk, the deployer is using mirage 3… I think this PR is exceeding on the time I wanted to allocate for this, haha.

**Comment by hannesm** [comment]
For the patch posted above, here is a fix for the mirage utility https://github.com/mirage/mirage/pull/1491/commits/16e82ffb481fb81ef70828facb1731bc07b82b97

**Comment by hannesm** [comment]
> Erk, the deployer is using mirage 3… I think this PR is exceeding on the time I wanted to allocate for this, haha.

That's a bit sad. Well, I don't quite understand the whole "CI" and "deployment" stuff of this repository anyways... Eventually someone will (hopefully) fix it. Even a local build doesn't work for me (as mentioned in #792).

**Comment by hannesm** [comment]
> Thanks for your changes, from what I can tell the issue is changes in 4.4.0. If you apply the following patch, it should work:
> 
> ```ocaml
> --- a/mirage/config.ml
> +++ b/mirage/config.ml
> @@ -202,10 +202,18 @@ let enable_metrics =
>    let doc = Key.Arg.info ~doc:"Enable metrics reporting" [ "metrics" ] in
>    Key.(create "metrics" Arg.(flag doc))
>  
> +type i0 = I0
> +let i0 = Functoria.Type.v I0
> +let no0 = Functoria.impl "Int" job
> +
> +type n1 = N1
> +let n1 = Functoria.Type.v N1
> +let noop1 = Functoria.impl "Set.Make" (job @-> job)
> +
>  let optional_monitoring time pclock stack =
>    if_impl (Key.value enable_metrics)
>      (mirage_monitoring $ time $ pclock $ stack)
> -    noop
> +    (noop1 $ no0)
>  
>  let () =
>    register "www"
> ```

I wanted to raise my voice that since mirage 4.4.2 got released, the above diff is not needed anymore. Not sure whether this makes any difference for this PR (which seems abandoned).

**Comment by MisterDA** [comment]
Hi Hannes,
I'm currently away from computers until the beginning of April. Unfortunately I was unable to set up a working build environment before I left, and thus complete this PR. If you feel you can take anything from it please do, otherwise I'll try to catch up in a month or so. 
-- Antonin

**Comment by samoht** [comment]
Thanks, some of your changes have been picked by https://github.com/mirage/mirage-www/pull/818 and now the CI is green again!

**Pull request closed without merging**